docgen/matrix_analytics_docgen/__main__.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+import glob
+from jinja2 import Environment, PackageLoader, select_autoescape
+import json
+import os.path
+import sys
+from typing import Optional, Union, Any
+
+
+@dataclass
+class EnumMember:
+    name: str
+    description: Optional[str]
+
+PropertyType = Union[str, list[EnumMember]]
+
+@dataclass
+class Property:
+    name: str
+    description: Optional[str]
+    type: PropertyType
+
+
+@dataclass
+class Event:
+    name: str
+    properties: list[Property]
+
+
+def load_schema(json: str) -> Event:
+    def load_property_type(d: dict) -> PropertyType:
+        if "oneOf" in d:
+            return [EnumMember(o["const"], o["description"]) for o in d["oneOf"]]
+        elif "enum" in d:            
+            return [EnumMember(name=e, description=None) for e in d["enum"]]
+        else:
+            return d["type"]
+
+    def load_property(name: str, d: dict[str, Any]) -> Property:
+        return Property(name, d.get("description"), load_property_type(d))
+
+    properties = [load_property(k, v) for k, v in json["properties"].items() if k != "eventName"]
+    return Event(name=json["properties"]["eventName"]["enum"][0], properties=list(properties))
```

Comment:
Is it possible to use the same code than with the file `mobile_generator.py`, and avoid code duplication?